src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/ExtractRelationMembersToNewRelationAction.java

```diff
@@ -0,0 +1,218 @@
+import org.openstreetmap.josm.tools.Logging;
+import org.openstreetmap.josm.tools.Utils;
+
+import static org.openstreetmap.josm.gui.MainApplication.*;
+
+/*
+Extracts selected members to a new route relation
+and substitutes them with this new route relation
+in the original relation
+
+ * @author Florian, Polyglot
+ */
+public class ExtractRelationMembersToNewRelationAction extends AbstractRelationEditorAction {
+    public ExtractRelationMembersToNewRelationAction(IRelationEditorActionAccess editorAccess) {
+        super(editorAccess, IRelationEditorUpdateOn.MEMBER_TABLE_SELECTION,
+            IRelationEditorUpdateOn.MEMBER_TABLE_CHANGE,
+            IRelationEditorUpdateOn.TAG_CHANGE);
+        new ImageProvider("bus").getResource().attachImageIcon(this);
+    }
+
+    @Override
+    public boolean isExpertOnly() {
+        return true;
+    }
+
+    @Override
+    protected void updateEnabledState() {
+        final boolean newEnabledState = !editorAccess.getSelectionTableModel().getSelection().isEmpty()
+            && Optional.ofNullable(editorAccess.getTagModel().get("type")).filter(it -> it.getValue().matches("route|superroute")).isPresent()
+            && !editorAccess.getEditor().isDirtyRelation();
+
+        putValue(SHORT_DESCRIPTION, (
+            newEnabledState
+                ? I18n.tr("Extract part of the route into new relation")
+                : I18n.tr("Extract into new relation (needs type=route/superroute tag and at least one selected relation member)")
+        ));
+        setEnabled(newEnabledState);
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent actionEvent) {
+        final MemberTableModel memberTableModel = editorAccess.getMemberTableModel();
+        IRelationEditor editor = editorAccess.getEditor();
+        final Relation originalRelation = editor.getRelation();
+
+        // save the current state, otherwise accidents happen
+        memberTableModel.applyToRelation(originalRelation);
+        // todo unfortunately there are still index out of bounds exceptions happening
+
+        final Collection<RelationMember> selectedMembers = memberTableModel.getSelectedMembers();
+
+        List<Command> commands = new ArrayList<>();
+
+        String title = I18n.tr("Extract part from relation?");
+        String question = I18n.tr(
+            "Do you want to create a new relation with the {0} selected members?",
+            selectedMembers.size()
+        );
+        List<Relation> parentRelations = new ArrayList<>(Utils.filteredCollection(originalRelation.getReferrers(), Relation.class));
+        parentRelations.removeIf(parentRelation -> !Objects.equals(parentRelation.get("type"), "superroute"));
+        final int numberOfParentRelations = parentRelations.size();
+        String relationSingularOrPlural = I18n.tr("relation");
+        if (numberOfParentRelations > 1) {
+            relationSingularOrPlural = I18n.tr("relations");
+        }
```

Comment:
For plurals `I18n.trn()` should be used. This ensures this can be translated correctly to languages with more than two or just different singular/plural variants.